src/main/java/org/opensearch/timeseries/rest/handler/AbstractTimeSeriesActionHandler.java

```diff
@@ -398,22 +408,207 @@ protected void validateTimeField(boolean indexingDryRun, ActionListener<T> liste
+     *     <li>For {@code RestRequest.Method.PUT}: Validates that the job is not already running before proceeding
+     *         with updating the configuration. It updates the configuration and manages the result index mapping
+     *         if necessary.</li>
+     *     <li>For {@code RestRequest.Method.POST}: Creates a new configuration. If a custom result index or alias is specified:
+     *         <ul>
+     *             <li>If flattening of the result index mapping is enabled, it initializes a flattened result index,
+     *                 sets up an ingest pipeline, and updates the flattened result index settings to bind the ingest pipeline
+     *                 with the flattened result index, enabling the writing of flattened nested fields into the flattened result index.</li>
+     *             <li>If flattening is not enabled, directly returns the creation response.</li>
+     *         </ul>
+     *         If no custom result index or alias is specified, returns the creation response directly.</li>
+     * </ul>
+     *
+     * <p><b>Notes:</b></p>
+     * <ul>
+     *     <li>If the configuration has a custom result index or alias and flattening is enabled,
+     *         the flattened result index name is suffixed with the detector ID in lowercase.</li>
+     *     <li>The ingest pipeline ID is uniquely generated based on the detector ID in lowercase.</li>
+     * </ul>
+     *
+     * <p><b>Exceptions:</b></p>
+     * <ul>
+     *     <li>If the {@code createConfigResponse} is of an unexpected type, which indicates create config call has failed,
+     *         then an {@link IllegalStateException} is thrown.</li>
+     * </ul>
      */
     protected void prepareConfigIndexing(boolean indexingDryRun, ActionListener<T> listener) {
         if (method == RestRequest.Method.PUT) {
-            handler
-                .confirmJobRunning(
-                    clusterService,
-                    client,
-                    id,
-                    listener,
-                    () -> updateConfig(id, indexingDryRun, listener),
-                    xContentRegistry
-                );
+            handlePutRequest(indexingDryRun, listener);
+        } else {
+            handlePostRequest(indexingDryRun, listener);
+        }
+    }
+
+    private void handlePutRequest(boolean indexingDryRun, ActionListener<T> listener) {
+        handler.confirmJobRunning(clusterService, client, id, listener, () -> {
+            handleFlattenResultIndexMappingUpdate(listener);
+            updateConfig(id, indexingDryRun, listener);
+        }, xContentRegistry);
+    }
+
+    private void handlePostRequest(boolean indexingDryRun, ActionListener<T> listener) {
+        createConfig(indexingDryRun, ActionListener.wrap(createConfigResponse -> {
+            if (shouldHandleFlattening(indexingDryRun, createConfigResponse)) {
+                IndexAnomalyDetectorResponse response = (IndexAnomalyDetectorResponse) createConfigResponse;
```

Comment:
This method is used by both forecasting and AD. You cannot assume it is an IndexAnomalyDetectorResponse.

Reply:
but this if condition will only be executed when flatten result index is set to true, which is a concept under AD. Also the `shouldHandleFlattening` is also check for this response type to ensure it will only be executed when the call is from AD instead of forecasting. Can I keep the current way of implementation?